response_sequence.go

```diff
@@ -0,0 +1,26 @@
+package hijack
+
+type seq[T any] struct {
+	sequence []T
+	index    int
+}
+
+type Response struct {
+	Data  string
+	Error error
+}
+
+func (s *seq[T]) GetNext() T {
```

Comment:
Can we return an error instead of panicking when it already iterated to the end?
Also would be nice to have a “HasNext” function

Reply:
the idea for this is to use it in the tests with the next call, if we return an error, and there was a valid assert that it should error, we might get false positive in test runs. That's why GetNext doesn't return anything.

What purpose would the `HasNext` solve?

Reply:
`HasNext` would tell you if it has reached to the end. This is how I would like to use it
```golang
for seq.HasNext() {
    item := seq.Next()
    doSomethingWith(item)
}
```

Reply:
no, I mean in terms of graphql mocking, I wouldn't care if it has next, no?

Reply:
I prefer a panic here, because if we expect 3 calls and set up a sequence for those 3, receiving a 4th call is unexpected, and it’s best to fail fast.

Reply:
chatted offline. while I’m not thrilled with the idea of panicking, I think it’s a sound solution